frontend/src/pages/FunctionalComponent/RecurringRunDetailsV2FC.tsx

```diff
@@ -0,0 +1,207 @@
+import { commonCss, padding } from 'src/Css';
+import { KeyValue } from 'src/lib/StaticGraphParser';
+import { formatDateString, enabledDisplayStringV2, errorToMessage } from 'src/lib/Utils';
+import { triggerDisplayString } from 'src/lib/TriggerUtils';
+
+export function RecurringRunDetailsV2FC(props: PageProps) {
+  const { updateBanner, updateToolbar } = props;
+  const [refresh, setRefresh] = useState(true);
+  const [toolbarState, setToolbarState] = useState<ToolbarProps>();
+  const recurringRunId = props.match.params[RouteParams.recurringRunId];
+  const Refresh = () => setRefresh(refreshed => !refreshed);
+
+  const { data: recurringRun, refetch: refetchRecurringRun } = useQuery<
+    V2beta1RecurringRun | undefined,
+    Error
+  >(
+    ['recurringRun'],
+    async () => {
+      try {
+        return await Apis.recurringRunServiceApi.getRecurringRun(recurringRunId);
+      } catch (err) {
+        const errorMessage = await errorToMessage(err);
+        updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message:
+            `Error: failed to retrieve recurring run: ${recurringRunId}.` +
+            (errorMessage ? ' Click Details for more information.' : ''),
+          mode: 'error',
+        });
+        return;
+      }
+    },
+    { enabled: !!recurringRunId, staleTime: 0, cacheTime: 0 },
+  );
+
+  const experimentId = recurringRun?.experiment_id!;
+  const { data: experiment } = useQuery<V2beta1Experiment | undefined, Error>(
+    ['experiment'],
+    async () => {
+      try {
+        return await Apis.experimentServiceApiV2.getExperiment(experimentId);
+      } catch (err) {
+        const errorMessage = await errorToMessage(err);
+        updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message:
+            `Error: failed to retrieve this recurring run's experiment.` +
+            (errorMessage ? ' Click Details for more information.' : ''),
+          mode: 'warning',
+        });
+        return;
+      }
+    },
+    { enabled: !!experimentId, staleTime: 0 },
+  );
+
+  useEffect(() => {
+    setToolbarState(getInitialToolbarState());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [recurringRun]);
+
+  useEffect(() => {
+    if (toolbarState) {
+      toolbarState.actions[ButtonKeys.ENABLE_RECURRING_RUN].disabled =
+        recurringRun?.status === V2beta1RecurringRunStatus.ENABLED;
+      toolbarState.actions[ButtonKeys.DISABLE_RECURRING_RUN].disabled =
+        recurringRun?.status !== V2beta1RecurringRunStatus.ENABLED;
+      toolbarState.pageTitle = recurringRun?.display_name!;
+      toolbarState.breadcrumbs = getBreadcrumbs(experiment);
+      updateToolbar(toolbarState);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [toolbarState, experiment, recurringRun]);
+
+  useEffect(() => {
+    refetchRecurringRun();
+  }, [refresh, refetchRecurringRun]);
```

Comment:
I think we should only run refresh when `refresh` is changed. I am not sure if calling `refetchRecurringRun()` every time when a `refetchRecurringRun` is changed will cause an infinite loop.